File: foodRecipe/account/accountManager.m
//
//  accountManager.m
//  foodRecipe
//
//  Created by Jin-young Ko on 2015. 9. 12..
//  Copyright (c) 2015년 Jin-young Ko. All rights reserved.
//

#import "accountManager.h"

@implementation accountManager

+ (UICKeyChainStore *) myKeyChain {
    return [UICKeyChainStore keyChainStoreWithService:@"MakCipe" accessGroup:@"com.MakSSe"];
}

+ (NSString *) getToken {
    return [[self myKeyChain] stringForKey:@"token"];
}

+ (void) setToken:(NSString *)token {
    [[self myKeyChain] setString:token forKey:@"token"];
}

+ (BOOL) hasToken {
    NSError *error;
    [[self myKeyChain] stringForKey:@"token" error:&error];
    if (error) {
        return NO;
    }
    
    return YES;
}

@end

File: foodRecipe/account/accountManager.swift
//
//  accountManager.swift
//  foodRecipe
//
//  Created by Jin-young Ko on 2015. 7. 29..
//  Copyright (c) 2015년 MakSSe. All rights reserved.
//

import Foundation
import KeychainAccess

class accountManager {
    
    static func myKeyChain() -> Keychain {
        return Keychain(service: "foodRecipe", accessGroup: "com.MakSSe")
    }
    
    static func getToken() -> String? {
        return myKeyChain().get("token")
    }
    
    static func setToken(token:String) {
        myKeyChain().set(token, key: "token")
    }
    
    static func hasToken() -> Bool {
        return getToken() == nil ? false : true
    }
}